src/saveload/oldloader_sl.cpp

```diff
@@ -499,7 +499,12 @@ static Town *RemapTown(TileIndex fallback)
 {
 	/* In some cases depots, industries and stations could refer to a missing town. */
 	Town *t = Town::GetIfValid(RemapTownIndex(_old_town_index));
-	if (t == nullptr) t = CalcClosestTownFromTile(fallback);
+	if (t == nullptr) {
+		/* In case the town that was refered to does not exist, find the closest.
+		 * However, this needs the kd-tree to be present. */
+		RebuildTownKdtree();
```

Comment:
One can argue if we should load it every time, but .. it is oldloader .. I don't actually care about performance :D

Reply:
Exactly my thoughts...